### PR TITLE
remove required package lanms for Windows

### PR DESCRIPTION
**Pull request by Hsiayukoo**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [✔] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ✔] Your code builds clean without any errors or warnings
- [ ✔] You are using approved terminology
- [✔ ] You have added unit tests

## Motivation

when I run 
```shell
pip install -r requirements.txt
```

image

it is failed because of  the package **lanms**, the package does not support the windows environment very well. So I read the origin paper : https://arxiv.org/pdf/1704.03155v2 ,and write the algorithm in pure Python.

image

and I also remain the lanms package for systems except Windows, the pure python based lanms only works in Windows.

## Test Plan

- check the lanms is ignored when pip install in windows

lanms被排除

- check the eval result is same as origin

my_lanms

- check the eval result is same as the official version lanms

the author also provide a python version lanms( but is can not used by using pip install )

https://github.com/argman/EAST/blob/master/locality_aware_nms.py

official_lanms

### CODE REVIEW

File: deploy/py_infer/src/data_process/postprocess/det_east_postprocess.py [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 import cv2
-import lanms
 import numpy as np
 
 # add mindocr root path, and import postprocess from mindocr
 mindocr_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../.."))
 sys.path.insert(0, mindocr_path)
 
 from mindocr.postprocess.det_base_postprocess import DetBasePostprocess  # noqa
+from mindocr.postprocess.nms_py.lanms_py import lanms_win
```
[the remark this answers is not in the document]
Comment (Hsiayukoo) [reply deploy/py_infer/src/data_process/postprocess/det_east_postprocess.py:13]: 这里包的名字是 lanms_py ，函数名是 lanms_win，要改一下名字咩
image

[the remark this answers is not in the document]
Comment (Mark-ZhouWX) [reply deploy/py_infer/src/data_process/postprocess/det_east_postprocess.py:13]: `try:
    from lanms import merge_quadrangle_n9
except ImportError:
    if platform.system() == "Windows":
        from mindocr.postprocess.nms_py.lanms_py import merge_quadrangle_n9
    else:
        raise ImportError("can not import lanms or lanms_win")``` 这样看起来更符合逻辑些

[the remark this answers is not in the document]
Comment (Hsiayukoo) [reply deploy/py_infer/src/data_process/postprocess/det_east_postprocess.py]: 已修改

**Comment by Hsiayukoo** [comment]
本次进行了如下验证
（1） v0.3.2 分支进行了静态图验证
（2）master 分支用 MindSpore 2.2.14 进行了动态图验证（因为 eval 脚本里面**存在 2.3 版本的特性**，而当前 MindSpore 版本 MindSpore 不支持 CPU，无法验证）
image

File: tools/dataset_converters/ic15.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def convert(self, task="det", image_dir=None, label_path=None, output_path=None)
 
     def _format_det_label(self, image_dir, label_dir, output_path):
         label_paths = sorted(glob.glob(os.path.join(label_dir, "*.txt")))
-        with open(output_path, "w") as out_file:
+        with open(output_path, "w", encoding="utf-8") as out_file:
```
Comment (Mark-ZhouWX) [inline tools/dataset_converters/ic15.py:39]: 为啥要修改这里呢，改变编码会影响linux下运行吗，有测过吗

**Review by Mark-ZhouWX — APPROVED** [review]

File: tools/dataset_converters/ic15.py (hunk 2 above)
Comment (Hsiayukoo) [reply tools/dataset_converters/ic15.py:39]: 已在 modelarts 上验证。
image

**Review by panshaowu — APPROVED** [review]

**Pull request merged**